test/sqllogictest/transform/filter_index.slt

```diff
@@ -78,7 +78,6 @@ explain plan for select * from foo, bar where foo.a = abs(bar.a) and foo.a = 3
 %2 =
 | Join %0 %1
 | | implementation = Differential %1 %0.()
-| | demand = (#0..#5)
```

Comment:
Same as in the other spot. This seems like it asks `Join` to compute and maintain additional columns, that we will not inspect. Is that .. because the finishing is hiding a project maybe, or .. what's the secret? :D

Reply:
If finishing does hide a projection, perhaps we want to forcibly extract it to allow the dataflow to be more efficient.